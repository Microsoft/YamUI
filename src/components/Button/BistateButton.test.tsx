/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { BistateButton, BistateButtonProps, BistateButtonState } from '.';

describe('<BistateButton />', () => {
  let component: ShallowWrapper<BistateButtonProps, BistateButtonState>;

  const getProps = () => {
    return {
      text: 'Joined',
      activeProps: {
        text: 'Leave',
      },
    };
  };

  beforeEach(() => {
    component = shallow(<BistateButton {...getProps()} />).dive();
  });

  it('matches its snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('when mouse enters', () => {
    beforeEach(() => {
      component.simulate('mouseEnter');
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    describe('when blurs', () => {
      beforeEach(() => {
        component.simulate('blur');
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('when mouse leaves', () => {
      beforeEach(() => {
        component.simulate('mouseLeave');
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });
  });

  describe('when focuses', () => {
    beforeEach(() => {
      component.simulate('focus');
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    describe('when mouse leaves', () => {
      beforeEach(() => {
        component.simulate('mouseLeave');
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('when blurs', () => {
      beforeEach(() => {
        component.simulate('blur');
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });
  });

  describe('with an onMouseEnter prop', () => {
    let onMouseEnter: jest.Mock;

    beforeEach(() => {
      onMouseEnter = jest.fn();
      component = shallow(<BistateButton {...getProps()} onMouseEnter={onMouseEnter} />);
    });

    describe('when the mouse enters', () => {
      beforeEach(() => {
        component.simulate('mouseEnter');
      });

      it('calls onMouseEnter', () => {
        expect(onMouseEnter).toBeCalled();
      });
    });
  });

  describe('with an onMouseLeave prop', () => {
    let onMouseLeave: jest.Mock;

    beforeEach(() => {
      onMouseLeave = jest.fn();
      component = shallow(<BistateButton {...getProps()} onMouseLeave={onMouseLeave} />);
    });

    describe('when the mouse leaves', () => {
      beforeEach(() => {
        component.simulate('mouseLeave');
      });

      it('calls onMouseLeave', () => {
        expect(onMouseLeave).toBeCalled();
      });
    });
  });

  describe('with an onFocus prop', () => {
    let onFocus: jest.Mock;

    beforeEach(() => {
      onFocus = jest.fn();
      component = shallow(<BistateButton {...getProps()} onFocus={onFocus} />);
    });

    describe('when focusing', () => {
      beforeEach(() => {
        component.simulate('focus');
      });

      it('calls onFocus', () => {
        expect(onFocus).toBeCalled();
      });
    });
  });

  describe('with an onBlur prop', () => {
    let onBlur: jest.Mock;

    beforeEach(() => {
      onBlur = jest.fn();
      component = shallow(<BistateButton {...getProps()} onBlur={onBlur} />);
    });

    describe('when blurring', () => {
      beforeEach(() => {
        component.simulate('blur');
      });

      it('calls onBlur', () => {
        expect(onBlur).toBeCalled();
      });
    });
  });
});
