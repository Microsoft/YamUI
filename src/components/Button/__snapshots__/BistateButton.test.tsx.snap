// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BistateButton /> matches its snapshot 1`] = `
<Button
  color="secondary"
  iconPosition="left"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  secondaryText="Leave"
  size="regular"
  status="enabled"
  text="Joined"
  type="button"
/>
`;

exports[`<BistateButton /> when focuses matches its snapshot 1`] = `
<Button
  color="secondary"
  iconPosition="left"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  secondaryText="Joined"
  size="regular"
  status="enabled"
  text="Leave"
  type="button"
/>
`;

exports[`<BistateButton /> when focuses when blurs matches its snapshot 1`] = `
<Button
  color="secondary"
  iconPosition="left"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  secondaryText="Leave"
  size="regular"
  status="enabled"
  text="Joined"
  type="button"
/>
`;

exports[`<BistateButton /> when focuses when mouse leaves matches its snapshot 1`] = `
<Button
  color="secondary"
  iconPosition="left"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  secondaryText="Joined"
  size="regular"
  status="enabled"
  text="Leave"
  type="button"
/>
`;

exports[`<BistateButton /> when mouse enters matches its snapshot 1`] = `
<Button
  color="secondary"
  iconPosition="left"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  secondaryText="Joined"
  size="regular"
  status="enabled"
  text="Leave"
  type="button"
/>
`;

exports[`<BistateButton /> when mouse enters when blurs matches its snapshot 1`] = `
<Button
  color="secondary"
  iconPosition="left"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  secondaryText="Joined"
  size="regular"
  status="enabled"
  text="Leave"
  type="button"
/>
`;

exports[`<BistateButton /> when mouse enters when mouse leaves matches its snapshot 1`] = `
<Button
  color="secondary"
  iconPosition="left"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  secondaryText="Leave"
  size="regular"
  status="enabled"
  text="Joined"
  type="button"
/>
`;
